Content.Server/ShipEvent/Systems/ShipEventFactionSystem.cs

```diff
@@ -0,0 +1,148 @@
+﻿using System.Linq;
+using Content.Server.Chat.Systems;
+using Content.Server.Ghost;
+using Content.Server.Ghost.Roles.Events;
+using Content.Server.Mind.Components;
+using Content.Server.Roles;
+using Content.Server.ShipEvent.Components;
+using Content.Server.Shuttles.Components;
+using Content.Shared.Shuttles.Components;
+
+namespace Content.Server.ShipEvent.Systems;
+
+public sealed class ShipEventFactionSystem : EntitySystem
+{
+    [Dependency] private readonly IEntityManager entMan = default!;
+    [Dependency] private readonly ChatSystem chatSystem = default!;
+
+    private int LastTeamNumber = 0;
+
+    /// <summary>
+    /// Dictionary with all the teams, where spawner uid is the key, and faction is the value
+    /// </summary>
+    public Dictionary<EntityUid, PlayerFaction>? Teams
+    {
+        get;
+        private set;
+    }
+    public override void Initialize()
+    {
+        base.Initialize();
+        Teams = new Dictionary<EntityUid, PlayerFaction>();
```

Comment:
```suggestion
    public Dictionary<EntityUid, PlayerFaction> Teams = new Dictionary<EntityUid, PlayerFaction>();
    {
        get;
        private set;
    }
    public override void Initialize()
    {
        base.Initialize();
```
можно же так сделать и убрать нуллы